Modules/@babylonjs/react-native/windows/BabylonReactNative/EngineViewManager.cpp

```diff
@@ -0,0 +1,148 @@
+#include "pch.h"
+#include "EngineViewManager.h"
+
+#include "JSValueReader.h"
+#include "JSValueXaml.h"
+
+using namespace winrt::Microsoft::ReactNative;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Media;
+using namespace winrt::Windows::UI::Xaml::Controls;
+using namespace winrt::Windows::UI::Xaml::Input;
+
+namespace winrt::BabylonReactNative::implementation {
+    EngineViewManager::EngineViewManager() {}
+
+    // IViewManager
+    hstring EngineViewManager::Name() noexcept {
+        return L"EngineView";
+    }
+
+    FrameworkElement EngineViewManager::CreateView() noexcept {
+        _swapChainPanel = SwapChainPanel();
+        _swapChainPanelPtr = nullptr;
+        copy_to_abi(_swapChainPanel, _swapChainPanelPtr);

```

Comment:
`copy_to_abi` ends up invoking `AddRef` on the underlying object. It looks like there's no corresponding place where we call `Release` meaning the swap chain panel is probably being leaked.

I'm not quite sure what you intend the ownership semantics of `_swapChainPanelPtr` to be. If you're only looking to get access to a `void*` to pass across an ABI boundary or something, but without changing the ref-count, I think it's fine to just use `get_abi`. If you're intending to increment the ref-count, make sure you're using something like `attach_abi` or something on the other side to ensure the lifetime is handled correctly.

Reply:
i think we should just use get_abi here and allow EngineViewManager to maintain SwapChainPanel lifecycle. thanks for the additional context!